### PR TITLE
9 translate the preferred client mode names

### PR DESCRIPTION
**Pull request by lukasmatusiewicz**
- Add translation
- Update tests

### CODE REVIEW

File: Tests/Tests.cs [hunk 1]
```diff
@@ -12,6 +13,8 @@ public class Tests
     {
         WireMockServer server;
         PrivacyIDEA privacyIDEA;
+        private readonly string authToken = "eyJ0eXAiOiJ...KV1QiLC6chGIM";
```
Comment (nilsbehlen) [inline Tests/Tests.cs:16]: _AuthToken

File: Tests/Tests.cs [hunk 2]
```diff
@@ -208,79 +261,13 @@ public void TriggerChallenges()
                     .WithPath("/validate/triggerchallenge")
                     .UsingPost()
                     .WithBody("user=test")
-                    .WithHeader("Authorization", authToken)
+                    .WithHeader("Authorization", this.authToken)
```
Comment (nilsbehlen) [inline Tests/Tests.cs:264]: will be _AuthToken. Can drop the `this`

File: Tests/Tests.cs [hunk 3]
```diff
@@ -290,32 +277,141 @@ public void TriggerChallenges()
...
+            Assert.AreEqual("otp", resp.PreferredClientMode);
+
+            // Test preferred_client_mode: webauthn
+            server
+            .Given(
+                Request.Create()
+                .WithPath("/validate/triggerchallenge")
+                .UsingPost()
+                .WithBody("user=testwebauthn")
+                .WithHeader("Authorization", this.authToken)
```
Comment (nilsbehlen) [inline Tests/Tests.cs:352]: see above

**Review by nilsbehlen — CHANGES_REQUESTED** [review]

File: Tests/Tests.cs (hunk 1 above)
Comment (lukasmatusiewicz) [reply Tests/Tests.cs]: [98fb81a](https://github.com/privacyidea/csharp-client/pull/10/commits/98fb81a65433bb4a424fab9883efcc50ad39cfd4)

File: Tests/Tests.cs (hunk 2 above)
Comment (lukasmatusiewicz) [reply Tests/Tests.cs]: [98fb81a](https://github.com/privacyidea/csharp-client/pull/10/commits/98fb81a65433bb4a424fab9883efcc50ad39cfd4)

File: Tests/Tests.cs (hunk 3 above)
Comment (lukasmatusiewicz) [reply Tests/Tests.cs]: [98fb81a](https://github.com/privacyidea/csharp-client/pull/10/commits/98fb81a65433bb4a424fab9883efcc50ad39cfd4)

**Comment by lukasmatusiewicz** [comment]
77.1% test coverage

**Comment by lukasmatusiewicz** [comment]
81.6% code coverage

**Comment by lukasmatusiewicz** [comment]
ready to review

File: PrivacyIDEA_Client/PrivacyIDEA.cs [hunk 4]
```diff
@@ -355,7 +355,11 @@ public void SetServiceAccount(string user, string pass, string realm = "")
         {
             _serviceUser = user;
             _servicePass = pass;
-            if (string.IsNullOrEmpty(realm) is false)
+            if (string.IsNullOrEmpty(realm))
```
Comment (nilsbehlen) [inline PrivacyIDEA_Client/PrivacyIDEA.cs:358]: _serviceRealm = realm ?? "";

File: Tests/TestTriggerChallenges.cs [hunk 5]
```diff
@@ -0,0 +1,703 @@
...
+        [TestCleanup]
+        public void Cleanup()
+        {
+            server.Stop();
+        }
+
+        [TestMethod]
+        public async Task TriggerChallenges()
+        {
+// Test no service account
```
Comment (nilsbehlen) [inline Tests/TestTriggerChallenges.cs:35]: Can you extract each of the comments to a function with that name

**Review by nilsbehlen — CHANGES_REQUESTED** [review]

File: Tests/TestTriggerChallenges.cs (hunk 5 above)
Comment (lukasmatusiewicz) [reply Tests/TestTriggerChallenges.cs]: [8d9d7b4](https://github.com/privacyidea/csharp-client/pull/10/commits/8d9d7b460d2fcb694b8bc2006585cc5541a0b129)

File: PrivacyIDEA_Client/PrivacyIDEA.cs (hunk 4 above)
Comment (lukasmatusiewicz) [reply PrivacyIDEA_Client/PrivacyIDEA.cs]: [26f9b42](https://github.com/privacyidea/csharp-client/pull/10/commits/26f9b42259abd9620ba06b0e0f21cc249415dc3f)

**Comment by lukasmatusiewicz** [comment]
85.9% code coverage

Ready for review

**Pull request merged**